### PR TITLE
Security - add variable to enable or not IMDSv2

### PR DESCRIPTION
**Pull request by jportasa**

### CODE REVIEW

File: modules/asg_node_group/variables.tf [hunk 1]
```diff
@@ -178,3 +178,9 @@ variable "bottlerocket_admin_container_source" {
   default     = ""
   description = "URI of a custom admin container image"
 }
+
+variable "imdsv2_enabled" {
```
Comment (errm) [inline modules/asg_node_group/variables.tf:182]: I think name like "required" or "enforced" would be more descriptive

Comment (jportasa) [reply modules/asg_node_group/variables.tf]: changed to imdsv2_required

Comment (jportasa) [reply modules/asg_node_group/variables.tf]: changed to imdsv2_required

**Review by piotr-zuralski — APPROVED** [review]

File: examples/cluster/bottlerocket_gpu_node_group/main.tf [hunk 2]
```diff
@@ -24,6 +24,9 @@ module "node_group" {
   instance_types = ["g4dn.xlarge"]
   min_size       = 1
 
+  imdsv2_required = true
```
Comment (errm) [inline examples/cluster/bottlerocket_gpu_node_group/main.tf:27]: do we need to set this everywhere in the examples - it is the default?

Comment (jportasa) [reply examples/cluster/bottlerocket_gpu_node_group/main.tf:27]: the default is true, I'll remove it from here just to make it cleaner

**Pull request merged**